fdbserver/storageserver.actor.cpp

```diff
@@ -3471,6 +3471,56 @@ ACTOR Future<GetRangeReqAndResultRef> quickGetKeyValues(
 	}
 };
 
+void unpackKeyTuple(Tuple** referenceTuple, Optional<Tuple> keyTuple, KeyValueRef* keyValue) {
+	if (!keyTuple.present()) {
+		// May throw exception if the key is not parsable as a tuple.
+		try {
+			keyTuple = Tuple::unpack(keyValue->key);
```

Comment:
`keyTuple` is of type `struct Tuple`, so the memory will be invalid after the `try` block. Thus, line 3484 is using invalid memory.

Reply:
I tried to do it the same way as it currently does now, https://github.com/apple/foundationdb/blame/main/fdbserver/storageserver.actor.cpp#L3527-L3533, will talk offline